graftm/hmmer.py

```diff
@@ -535,136 +491,226 @@ def extract_orfs(self, input_path, raw_orf_path, hmmsearch_out_path, orf_titles_
         searcher.hmmsearch(cmd, self.search_hmm, output_table_list)
         
         with open(orf_titles_path, 'w') as output:
-            seen = []
-            for table in output_table_list:
-                for title in [x.split(' ')[0] for x in open(table).readlines() if not x.startswith('#')]:
-                    if title not in seen:
-                        output.write(str(title) + '\n')
-                        seen.append(title)
-                    else:
-                        pass       
-        
+
+            reads= set(
+                       list(
+                            itertools.chain(
+                                            *[HMMreader(x).names() for x \
+                                              in output_table_list]
+                                            )
+                            )
+                       )
+            [output.write(x + '\n') for x in reads]
+            
         # Extract the reads using the titles.
         cmd = 'fxtract -H -X -f %s %s > %s' % (orf_titles_path, raw_orf_path, orf_out_path)
         
         logging.debug("Running command: %s" % cmd)
         subprocess.check_call(cmd, shell=True)
+    
+    def _get_read_names(self, search_result, r):
+        '''
+        _get_read_names - loops through hits and alingment spans of hmm hits
+        and determines potentially linked hits (for example, if a gene in a 
+        contig hits a hmm more than once, in two different conserved regions
+        within a gene) and combines them into one 'hit'. The total span of the
+        hits deemed to be linked is returned.
         
-        # Return name of output file
-        return orf_out_path
-
-    def p_search(self, files, args, run_stats, base, unpack, raw_reads):
-        '''Protein search pipeline - The searching step for the protein 
-        pipeline, where hits are identified in the input reads through 
-        HMMsearches
+        Parameters
+        ----------
+        search_result : obj
+            SequenceSearchResult object with all paramaters defined. Used here
+            to create rows containing information on alignment direction and 
+            alignment span.
+        r : int
```

Comment:
use real words for variables. Saves time in the long run, and really isn't that hard to type out